src/dag/write.rs

```diff
@@ -28,11 +28,14 @@ impl<'a> Write<'_> {
         Ok(())
     }
 
-    pub async fn set_head(&mut self, name: &str, hash: &str) -> Result<()> {
-        Ok(self
-            .kvw
-            .put(&Key::Head(name).to_string(), hash.as_bytes())
-            .await?)
+    pub async fn set_head(&mut self, name: &str, hash: Option<&str>) -> Result<()> {
+        match hash {
+            None => Ok(self.kvw.del(&Key::Head(name).to_string()).await?),
```

Comment:
Dammit, I write this same bug every time I work on a system like this.

Reply:
We could also do `del_head`? What's more natural for caller code?

Reply:
when it comes to utilities i often prefer to have smaller caller-composable pieces